vcd/resource_vcd_vapp_vm_tools.go

```diff
@@ -0,0 +1,1168 @@
+		updateNeeded = true
+	}
+
+	if memorySharesLevel, ok := d.GetOk("cpu_priority"); ok {
+		vmSpecSection.CpuResourceMhz.SharesLevel = memorySharesLevel.(string)
+		updateNeeded = true
+	}
+
+	if memoryLimit, ok := d.GetOk("cpu_limit"); ok {
+		vmSpecSection.CpuResourceMhz.Limit = takeInt64Pointer(int64(memoryLimit.(int)))
+		updateNeeded = true
+	}
+
+	if memoryShares, ok := d.GetOk("cpu_shares"); ok {
+		vmSpecSection.CpuResourceMhz.Shares = takeIntPointer(memoryShares.(int))
+		updateNeeded = true
+	}
+
+	if memoryReservation, ok := d.GetOk("cpu_reservation"); ok {
+		vmSpecSection.CpuResourceMhz.Reservation = takeInt64Pointer(int64(memoryReservation.(int)))
+		updateNeeded = true
+	}
+
+	if updateNeeded {
+		err := updateVmSpecSection(vmSpecSection, vm, description)
+		if err != nil {
+			return fmt.Errorf("error updating advanced compute settings: %s", err)
+		}
+	}
+
+	// Refresh VM to ensure that latest VM structure is used in other function calls
+	err := vm.Refresh()
+	if err != nil {
+		return fmt.Errorf("error refreshing VM after updating advanced compute settings: %s", err)
+	}
+
+	return nil
+}
+
+// isItVappNetwork checks if it is an vApp network (not vApp Org Network)
+func isItVappNetwork(vAppNetworkName string, vapp govcd.VApp) (bool, error) {
+	vAppNetworkConfig, err := vapp.GetNetworkConfig()
+	if err != nil {
+		return false, fmt.Errorf("error getting vApp networks: %s", err)
+	}
+
+	for _, networkConfig := range vAppNetworkConfig.NetworkConfig {
+		if networkConfig.NetworkName == vAppNetworkName &&
+			govcd.IsVappNetwork(networkConfig.Configuration) {
+			log.Printf("[TRACE] vApp network found: %s", vAppNetworkName)
+			return true, nil
+		}
+	}
+
+	return false, fmt.Errorf("configured vApp network isn't found: %s", vAppNetworkName)
+}
+
+type diskParams struct {
+	name       string
+	busNumber  *int
+	unitNumber *int
+}
+
+func expandDisksProperties(v interface{}) ([]diskParams, error) {
+	v = v.(*schema.Set).List()
+	l := v.([]interface{})
+	diskParamsArray := make([]diskParams, 0, len(l))
+
+	for _, raw := range l {
+		if raw == nil {
+			continue
+		}
+		original := raw.(map[string]interface{})
+		addParams := diskParams{name: original["name"].(string)}
+
+		busNumber := original["bus_number"].(string)
+		if busNumber != "" {
+			convertedBusNumber, err := strconv.Atoi(busNumber)
+			if err != nil {
+				return nil, fmt.Errorf("value `%s` bus_number is not number. err: %s", busNumber, err)
+			}
+			addParams.busNumber = &convertedBusNumber
+		}
+
+		unitNumber := original["unit_number"].(string)
+		if unitNumber != "" {
+			convertedUnitNumber, err := strconv.Atoi(unitNumber)
+			if err != nil {
+				return nil, fmt.Errorf("value `%s` unit_number is not number. err: %s", unitNumber, err)
+			}
+			addParams.unitNumber = &convertedUnitNumber
+		}
+
+		diskParamsArray = append(diskParamsArray, addParams)
+	}
+	return diskParamsArray, nil
+}
+
+func getVmIndependentDisks(vm govcd.VM) []string {
```

Comment:
Please add a comment explaining what the return strings are

Reply:
Added